src/server/scripts/Northrend/Naxxramas/boss_grobbulus.cpp

```diff
@@ -101,22 +123,65 @@ struct boss_grobbulus : public BossAI
     }
 };
 
-struct npc_grobbulus_poison_cloud : public ScriptedAI
+struct npc_grobbulus_poison_cloud : public NullCreatureAI
 {
-    npc_grobbulus_poison_cloud(Creature* creature) : ScriptedAI(creature)
+    npc_grobbulus_poison_cloud(Creature* creature) : NullCreatureAI(creature) { }
+
+    void JustAppeared() override
+    {
+        DoCastSelf(SPELL_POISON_CLOUD_PASSIVE);
+        DoCastSelf(SPELL_PACIFY_SELF);
+        me->DespawnOrUnsummon(61s);
+    }
+};
+
+struct npc_fallout_slime : public ScriptedAI
+{
+    npc_fallout_slime(Creature* creature) : ScriptedAI(creature)
     {
-        SetCombatMovement(false);
-        creature->SetReactState(REACT_PASSIVE);
+        Initialize();
     }
 
-    void IsSummonedBy(WorldObject* /*summoner*/) override
+    void Initialize()
```

Comment:
IMO JustAppeared is a better fit for these initializations, you could drop the Initialize method altogether

Reply:
>  better fit for these initializations

Even for SetCorpseDelay?

Reply:
you can set corpse delay anywhere, even in the ctor of the script. there is like no standard for it at all. JustAppeared() is not really a good spot for such initializations because it's being called after the first update tick of the creature instead of right at spawn. InitializeAI would be the correct spot for immediate changes.

Reply:
ummm true, JustAppeared is delayed one tick

Reply:
eitherway, IMO, Casts feel like they should be handled on JustAppeared
dont know, it really depends on each scenario